openedx/features/genplus_features/genplus/rmunify.py

```diff
@@ -91,12 +102,82 @@ def fetch_students(self, query=Class.visible_objects.all()):
             gen_user_ids = []
             for student_data in data['Students']:
                 student_email = student_data.get('UnifyEmailAddress')
+                identity_guid = student_data.get('IdentityGuid')
                 gen_user, created = GenUser.objects.get_or_create(
                     email=student_email,
                     role=GenUserRoles.STUDENT,
                     school=gen_class.school,
                 )
+                # update the identity_guid
+                gen_user.identity_guid = identity_guid
+                gen_user.save()
                 gen_user_ids.append(gen_user.pk)
 
             gen_students = Student.objects.filter(gen_user__in=gen_user_ids)
             gen_class.students.add(*gen_students)
+
+
+class RmUnifyProvisioning(BaseRmUnify):
+
+    UPDATES = '/appprovisioning/v2/{}/updates/'
+    DELETE_BATCH = '/appprovisioning/v2/{}/deletebatch/'
+
+    def get_header(self):
+        return {"Authorization": "Unify " + self.timestamp + ":" + self.hashed}
+
+    def provision(self):
+        data = self.fetch(self.UPDATES.format(self.key), provisioning=True)
+        if data:
+            updates_batch = []
+            # check updates and update/delete user accordingly
+            for update in data['Updates']:
+                if update['Type'] == RmUnifyUpdateTypes.USER:
+                    self.update_user(update['UpdateData'])
+                elif update['Type'] == RmUnifyUpdateTypes.DELETE_USER:
+                    try:
+                        # only deleting if user with unify guid exist in our system
+                        identity_guid = update['UpdateData']['IdentityGuid']
+                        genplus_tasks.delete_user.apply_async(
```

Comment:
No need to make an async task for this because provisioning will be handled by a cron job.